src/locale.js

```diff
@@ -76,23 +76,40 @@ export default class Locale {
 		this.contentLanguageDirection = getLanguageDirection( this.contentLanguage );
 
 		/**
-		 * Translates the given string to the {@link #uiLanguage}. This method is also available in
+		 * Translates the given message to the {@link #uiLanguage}. This method is also available in
 		 * {@link module:core/editor/editor~Editor#t} and {@link module:ui/view~View#t}.
 		 *
-		 * The strings may contain placeholders (`%<index>`) for values which are passed as the second argument.
+		 * This method's context is statically bound to the `Locale` instance and **always should be called as a function**:
+		 *
+		 *		const t = locale.t;
+		 *		t( 'Label' );
+		 *
+		 * The message can be either a string or an object implementing the {@link module:translation-service~Message} interface.
+		 *
+		 * A Message may contain placeholders (`%<index>`) for values that are passed as the second argument.
 		 * `<index>` is the index in the `values` array.
 		 *
-		 *		editor.t( 'Created file "%0" in %1ms.', [ fileName, timeTaken ] );
+		 *		t( 'Created file "%0" in %1ms.', [ fileName, timeTaken ] );
 		 *
-		 * This method's context is statically bound to Locale instance,
-		 * so it can be called as a function:
+		 * A Message can provide a plural form using the `plural` property and a value - that should be always the first element
+		 * of the `values` array based on which the plural form of the target language should be picked. That property value will
+		 * be used as a default plural translation when the translation for the target language will be missing.
 		 *
-		 *		const t = this.t;
-		 *		t( 'Label' );
+		 *		t( { string: 'Add a space', plural: 'Add %0 spaces' }, [ spaces ] );
+		 *		t( { string: '%1 a space', plural: '%1 %0 spaces' }, [ spaces, 'Add' ] );
+		 *
+		 * A Message can provide a context using the `context` property when the message string may be not enough unique
```

Comment:
You can't be "enough unique". You either are, or not. So, keep it simple: "when the message string ids not unique".